### PR TITLE
Update docfx sanity check

### PR DESCRIPTION
**Pull request by yufeih**
## Changes

Update docfx sanity check to use the latest version, which runs natively on linux and and is capable of more validations.

Fixes a broken anchor and a broken HTML tag problem detected by the new tool.

### CODE REVIEW

**Comment by yufeih** [comment]
There is one error detected I'm not sure how to update. [This file](https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/library-guidelines.md) contains a link to `README.md` which isn't part of the website, I guess it is a trick to link to the parent directory, but there isn't any API specifications page. The `API specifications` link seems to point to the wrong location on [this page](https://opentelemetry.io/docs/reference/specification/library-guidelines/)

File: .github/workflows/checks.yaml [hunk 1]
```diff
@@ -61,17 +61,19 @@ jobs:
       run: make misspell
 
   docfx:
-    runs-on: windows-latest
+    runs-on: unbuntu-latest
```
Comment (arminru) [inline .github/workflows/checks.yaml:64]:
```suggestion
    runs-on: ubuntu-latest
```

**Review by reyang — APPROVED** [review]

File: .github/workflows/checks.yaml [hunk 2]
```diff
@@ -61,17 +61,19 @@ jobs:
...
     steps:
     - name: check out code
       uses: actions/checkout@v2
 
     - name: install docfx
-      run: choco install docfx -y
+      uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 6.0.x
+      run: dotnet tool update -g docfx --version "3.0.0-*" --add-source https://docfx.pkgs.visualstudio.com/docfx/_packaging/docs-public-packages/nuget/v3/index.json
```
Comment (arminru) [inline .github/workflows/checks.yaml:73]:
```
Error : .github#L1
a step cannot have both the `uses` and `run` keys
```
See the Checks tab above (https://github.com/open-telemetry/opentelemetry-specification/pull/2285/checks).

Comment (yufeih) [reply .github/workflows/checks.yaml]: Thanks! yml file fixed.

**Comment by yufeih** [comment]
I've updated the check and it should pass. The checks needs a maintainer approval.

**Comment by yufeih** [comment]
The `markdown-lint` failure seems not related to this PR.

**Comment by arminru** [comment]
Hi @yufeih! The markdownlint issue should have been resolved in the meantime, let's give it another try.

File: .github/workflows/checks.yaml [hunk 3]
```diff
@@ -61,17 +61,18 @@ jobs:
...
 
     - name: install docfx
-      run: choco install docfx -y
+      run: dotnet tool update -g docfx --version "3.0.0-*" --add-source https://docfx.pkgs.visualstudio.com/docfx/_packaging/docs-public-packages/nuget/v3/index.json
 
-    - name: run .\docfx.cmd
-      shell: cmd
-      run: .\docfx.cmd
+    - name: run docfx
+      run: docfx build --dry-run
```
Comment (arminru) [inline .github/workflows/checks.yaml:75]: Will this fail the build if any issues come up or does it only print to them to the log in dry run mode?

Comment (yufeih) [reply .github/workflows/checks.yaml:75]: It'll fail the build because of non-zero exit code on error.

File: .github/workflows/checks.yaml [hunk 4]
```diff
@@ -61,17 +61,18 @@ jobs:
...
-    - name: check out code
-      uses: actions/checkout@v2
+    - uses: actions/checkout@v2
+    - uses: actions/setup-dotnet@v1
+      with:
+        dotnet-version: 6.0.x
 
     - name: install docfx
-      run: choco install docfx -y
+      run: dotnet tool update -g docfx --version "3.0.0-*" --add-source https://docfx.pkgs.visualstudio.com/docfx/_packaging/docs-public-packages/nuget/v3/index.json
```
Comment (arminru) [inline .github/workflows/checks.yaml:72]: Will `3.0.0-*` also match `3.0.0` once it's released or just versions with a `-suffix`?

**Review by arminru — APPROVED** [review]
LGTM, thanks @yufeih!

File: .github/workflows/checks.yaml (hunk 4 above)
Comment (arminru) [reply .github/workflows/checks.yaml:72]: @yufeih ping

Comment (yufeih) [reply .github/workflows/checks.yaml:72]: It'll be a version with a `-suffix`, we probably need to update this line to fix the version and remove the `--add-source` after the official release.

Comment (arminru) [reply .github/workflows/checks.yaml:72]: Alright, I added #2438 for this.

**Pull request merged**